src/vs/workbench/browser/parts/titlebar/titlebarPart.ts

```diff
@@ -182,10 +222,33 @@ export class BrowserTitleService extends MultiWindowParts<BrowserTitlebarPart> i
 	private variables: ITitleVariable[] = [];
 
 	registerVariables(variables: ITitleVariable[]): void {
-		this.variables.push(...variables);
+		const unregisteredVariables: ITitleVariable[] = [];
+
+		// Filter out any variables that are already registered
+		const existingVariableNames = new Set(this.variables.map(v => v.name));
+		const existingVariableContexKeys = new Set(this.variables.map(v => v.contextKey));
+		for (const v of variables) {
```

Comment:
This would be so much simpler if we would simply store the variables in a `Map` already mapped to the name of it?